controller/backup_volume_controller.go

```diff
@@ -289,6 +289,29 @@ func (bvc *BackupVolumeController) reconcile(backupVolumeName string) (err error
 		// Skip the Backup CR which is created from the local cluster and
 		// the snapshot backup hasn't be completed or pulled from the remote backup target yet
 		if b.Spec.SnapshotName != "" && b.Status.State != longhorn.BackupStateCompleted {
+			if b.Status.State == longhorn.BackupStateError || b.Status.State == longhorn.BackupStateUnknown {
+				// Start to delete the failed backups, default: true
+				autoFailedBackupCleanup, err := bvc.ds.GetSettingAsBool(types.SettingNameFailedBackupAutoDeleteion)
```

Comment:
Should this be moved outside the for loop so we don't repeatedly get the same setting value for each iteration?

Reply:
Moved before the loop.